processing/src/main/java/org/apache/druid/query/metadata/metadata/ColumnAnalysis.java

```diff
@@ -47,6 +50,7 @@ public static ColumnAnalysis error(String reason)
 
   @JsonCreator
   public ColumnAnalysis(
+      @JsonProperty("typeSignature") ColumnType typeSignature,
```

Comment:
can this change cause a problem during upgrade? E.g. an historical is on older version while broker is on newer version so the broker gets old json from the historical when running segment metadata queries?

Reply:
the supported upgrade order is brokers last, but it would just be null, which shouldn't be a problem i think

Reply:
https://github.com/apache/druid/pull/11895/files#diff-5aa1494dc5a13e82654e42ce8cdb3cb628c95233f8828312b3ede40dac1b8544R156 - shall this be modified to pick the one if the other is null instead of throwing an error. I am throwing darts in the dark here so please ignore if this is not really a problem :)

Reply:
if we supported updating the broker before data servers then we might wan't to consider that, but we don't (so that new query types, filters, aggregators, etc cannot be issued before data servers can understand them)

Reply:
ah yes. you are right. thanks for clarifying.